zips/zip-0312.rst

```diff
@@ -141,11 +141,56 @@ concretely in the Ciphersuites section.
 Key Generation
 --------------
 
-While key generation is out of scope for this ZIP and the FROST spec [#FROST]_,
-it needs to be consistent with FROST, see [#frost-tdkg]_ for guidance. The spend
-authorization private key :math:`\mathsf{ask}` [#protocol-spendauthsig]_ is the
-particular key that must be used in the context of this ZIP. Note that the
-:math:`\mathsf{ask}` is usually derived from the spending key
+While FROST key generation is out of scope for this ZIP and the FROST spec
+[#FROST]_, it needs to be consistent with FROST; see [#frost-tdkg]_ for general
+guidance. This section assumes all participants have run either a trusted dealer
+or distributed key generation process, and each participant has their signing
+key share `sk_i` and the group public key `PK`. Trusted dealer key generation
+MAY be used; distributed key generation SHOULD be used.
+
+To define a spending or viewing key that uses FROST, the Sapling and Orchard key
+trees [#protocol-saplingkeycomponents]_ [#protocol-orchardkeycomponents]_ are
+adjusted as follows:
+
+- The Spend validating key :math:`\mathsf{ak}` is replaced by the FROST group
+  public key `PK` [#frost-protocol]_.
+- The Spend authorizing key :math:`\mathsf{ask}` is replaced by the logical
+  signing key that corresponds to the group public key `PK`. By design, this key
+  is never explicitly constructed (unless trusted dealer key generation was
+  used), and instead is represented by any sufficient subset of the participant
+  FROST signing key shares `sk_i`.
+
+The remaining parts of the Sapling and Orchard key trees SHOULD be generated
+from a common spending key :math:`\mathsf{sk}` as described in
+[#protocol-saplingkeycomponents]_ [#protocol-orchardkeycomponents]_, with the
+exception of :math:`\mathsf{ask}` which SHOULD NOT be derived from
+:math:`\mathsf{sk}` (and can't possibly be derived, if distributed key
+generation was used). Deriving :math:`\mathsf{ask}` from :math:`\mathsf{sk}` MAY
+be done if the application requires to backup the unsplit key, but this is
+generally not recommended since it forces the usage of trusted dealer key
+generation.
```

Comment:
This is equivocating between SHOULD NOT and MAY for the same possibility. (Note that NOT RECOMMENDED in caps means the same thing as SHOULD NOT.)